### PR TITLE
Bug 1913341: [OCPCLOUD-1072] Add Disruptive test selection and mark proxy tests as disruptive.

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Proxy tests are containing some basic post condition identifying cluster as recovered.

### CODE REVIEW

**Review by michaelgugino — APPROVED** [review]
/lgtm

File: Makefile
Comment (JoelSpeed) [inline Makefile:74]: Which takes precedence here? Skip or Focus?

File: Makefile
Comment (JoelSpeed) [inline Makefile:71]: Should we be more explicit here and make this `[Disruptive]` including the brackets, just in case a test randomly has disruptive in the title anyway?

File: Makefile
Comment (Danil-Grigorev) [reply Makefile:74]: Definitely skip.

File: Makefile
Comment (Danil-Grigorev) [reply Makefile:71]: I'm afraid of possibility for this to become a wildcard regular expression.

Comment (JoelSpeed) [reply Makefile:71]: Does the syntax make it a regex, In which case we can escape the brackets right?

Comment (Danil-Grigorev) [reply Makefile]: Let's try this. Need another LGTM then.

**Comment by JoelSpeed** [comment]
/approve
/lgtm

Seems to be doing what it says on the tin

**Comment by Danil-Grigorev** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-api-actuator-pkg/204/pull-ci-openshift-cluster-api-actuator-pkg-master-e2e-azure-operator/1349735203181957120 is failing, the problem still persists.

/retest

**Comment by Danil-Grigorev** [comment]
Wait, I was assuming this proxy change got merged, and didn't take an effect, so marked webhook tests as disruptive. @JoelSpeed I did a change, should I revert it and will you be able to give it a second LGTM?..

File: Makefile
Comment (JoelSpeed) [inline Makefile:65]: Does this currently run any tests? Is it worth merging it uncommented and just having it be a no-op for now?

Comment (Danil-Grigorev) [reply Makefile:65]: I think we can uncomment it, but revert to the state of https://github.com/openshift/cluster-api-actuator-pkg/commit/d8f139d9f67dfd10dff4b7cfe3ea60c9cefba429 and https://github.com/openshift/cluster-api-actuator-pkg/commit/e3a1a4fb39a61566087d97d2139cadebf9482c87 first. Let'd mark proxy tests as disruptive for now, and consider adding post/pre conditions later @JoelSpeed

Comment (JoelSpeed) [reply Makefile:65]: I don't really follow what you mean by reverting? As far as I can tell, this should be good to merge if we just uncomment this line. What are you concerned about in particular?

Comment (Danil-Grigorev) [reply Makefile]: No, just trying to recall conversation in slack on this topic. SGTM and updated, PTAL @JoelSpeed

**Comment by JoelSpeed** [comment]
Code changes look ok to me, want to see how the tests pass/fail before giving approval

**Comment by Danil-Grigorev** [comment]
Previous code changes are sufficient to fix the issue mentioned in https://bugzilla.redhat.com/show_bug.cgi?id=1913341#c4 Reverting this to https://github.com/openshift/cluster-api-actuator-pkg/commit/d8f139d9f67dfd10dff4b7cfe3ea60c9cefba429

File: Makefile
Comment (JoelSpeed) [inline Makefile:64]: Should this use the same focus as the others skip?
```suggestion
	hack/ci-integration.sh $(GINKGO_ARGS) -focus="\[Disruptive\]" || (hack/junitmerge.sh && exit 1)
```

File: pkg/framework/framework.go
Comment (JoelSpeed) [inline pkg/framework/framework.go:133]: Do we need two helpers with different names like this? This to me implies that one is sync and one is async, but they are both async. Can we make this clearer somehow?

Comment (Danil-Grigorev) [reply pkg/framework/framework.go:133]: It is essentially like this. One scenario is expecting an absence of interruption in MAO `IsStatusAvailable` with minimal amount of polling (just to exclude case when is something with API server), and another is actually waiting for Available condition in ~3 min, which is the time required to roll out our components and verify that they are stable.

File: Makefile
Comment (Danil-Grigorev) [reply Makefile]: Fixed, PTAL.

File: pkg/framework/framework.go
Comment (JoelSpeed) [reply pkg/framework/framework.go:133]: Why not name them `WaitForStatusAvailableShort`/`WaitForStatusAvailableLong` in that case? I still think it's confusing that these are basically the same but have different names completely

Comment (Danil-Grigorev) [reply pkg/framework/framework.go]: Updated PTAL @JoelSpeed

File: pkg/framework/framework.go
Comment (JoelSpeed) [inline pkg/framework/framework.go:131]: Since this uses wait medium let's call it medium
```suggestion
func WaitForStatusAvailableMedium(client runtimeclient.Client, name string) bool {
```

**Comment by Danil-Grigorev** [comment]
/hold

Putting a hold on that, still needs some work. The post condition for proxy was not sufficient. Webhooks are still flaking ATM.

**Comment by JoelSpeed** [comment]
Why are we waiting now for KAS and KCM to become available?

**Comment by Danil-Grigorev** [comment]
> Why are we waiting now for KAS and KCM to become available?

This is the only way to ensure the cluster is healthy again. Or we will see flakes in our tests in different places.

**Comment by lobziik** [comment]
Might it make sense to have a separate make target and CI step as well, like we have for origin tests?
https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-master-presubmits.yaml#L962

**Pull request merged**
